crates/rust-analyzer/src/caps.rs

```diff
@@ -100,19 +131,24 @@ fn code_action_capabilities(client_caps: &ClientCapabilities) -> CodeActionProvi
         .as_ref()
         .and_then(|it| it.code_action.as_ref())
         .and_then(|it| it.code_action_literal_support.as_ref())
-        .map_or(CodeActionProviderCapability::Simple(true), |_| {
+        .map_or(CodeActionProviderCapability::Simple(true), |caps| {
+            let mut action_kinds = vec![
+                CodeActionKind::EMPTY,
+                CodeActionKind::QUICKFIX,
+                CodeActionKind::REFACTOR,
+                CodeActionKind::REFACTOR_EXTRACT,
+                CodeActionKind::REFACTOR_INLINE,
+                CodeActionKind::REFACTOR_REWRITE,
+            ];
+
+            // Not all clients can fall back gracefully for unknown values.
+            // Microsoft.VisualStudio.LanguageServer.Protocol.CodeActionKind does not support CodeActionKind::EMPTY
+            // So have to filter out.
+            action_kinds
```

Comment:
The spec says:

> codeActionKind: {

			/**
			 * The code action kind values the client supports. When this
			 * property exists the client also guarantees that it will
			 * handle values outside its set gracefully and falls back
			 * to a default value when unknown.
			 */
			valueSet: CodeActionKind[];
		};

I think the VS implementation is buggy and would prefer not having to workaround it.